### PR TITLE
tests(e2e): enable e2e tests in GitHub Actions

### PR DESCRIPTION
**Pull request by GabrielNagy**
This PR fixes the remaining issues with e2e tests and enables them to run on pushes to main. Workflow dispatch is also available where developers can trigger ad-hoc tests with custom adsys forks or refs.

Tests on Noble are currently disabled because of issues with the Azure templates. There hasn't been a new template for almost 1 month (strange) so I will come back to this once we have something new to work with.

Passing runs of the workflow on the other codenames can be seen here:
- triggered on push to main: https://github.com/GabrielNagy/adsys/actions/runs/8110510671
- triggered ad-hoc with custom parameters: https://github.com/GabrielNagy/adsys/actions/runs/8110696948
  + test only on Mantic
  + set repository to `GabrielNagy/adsys`
  + set branch to `e2e-tests-ci`

NB: For the `ADSYS_PRO_TOKEN` I'm using my personal Pro token, I don't know if we can get a bespoke token to use specifically for E2E tests.

Fixes UDENG-1536

### CODE REVIEW

**Review by denisonbarbosa — APPROVED** [review]
Nothing to say besides nice work!!!!

File: e2e/cmd/run_tests/01_provision_client/main.go
Comment (didrocks) [inline e2e/cmd/run_tests/01_provision_client/main.go:193]: hum, shouldn’t we fail quickly if we can’t install?

File: e2e/scripts/Dockerfile.build
Comment (didrocks) [inline e2e/scripts/Dockerfile.build:1]: can’t set the default to "latest" (will be easier in case of a manual run?

File: e2e/scripts/patches/mantic.patch
Comment (didrocks) [inline e2e/scripts/patches/mantic.patch:1]: ahah, I won’t be tricked this time that this is a symlink!

**Review by didrocks — COMMENTED** [review]
I’m just commenting on some considerations we might want to take before giving the final approval! Those are all opened questions :)

**Comment by didrocks** [comment]
(oh and of course, fixing the conflicts in the yaml file :))

File: e2e/cmd/run_tests/01_provision_client/main.go
Comment (didrocks) [reply e2e/cmd/run_tests/01_provision_client/main.go:193]: shouldn’t we log them at least, so that if some policy fails where we expected to have them working, we can see that was because of the package failing to install for $reasons?

Comment (didrocks) [reply e2e/cmd/run_tests/01_provision_client/main.go]: Ah, that’s perfect then! :)

**Pull request merged**
